ecl/hqlcpp/hqlres.cpp

```diff
@@ -313,13 +317,16 @@ void ResourceManager::addManifestsFromArchive(IPropertyTree *archive, ICodegenCo
         const char *xml = manifestContents;
         bool isSigned = false;
         // Check for signature
-        if (startsWith(xml, "-----BEGIN PGP SIGNED MESSAGE-----"))
+        if (queryCodeSigner().hasSignature(xml))
         {
             try
             {
-                OwnedHqlExpr sig = checkSignature(manifestContents.length(), manifestContents.str());
-                xml = stripSignature(strippedManifestContents, manifestContents).str();
-                isSigned = true;
+                StringBuffer signer, errmsg;
+                isSigned = queryCodeSigner().verifySignature(manifestContents, signer, errmsg);
+                if (isSigned)
+                    OwnedHqlExpr sig = createExprAttribute(_signed_Atom,createConstant(signer.str()));
```

Comment:
Similar, what's the purpose of having variable sig?

Reply:
This is not necessary. Removed.